ldcbench.http-node/src/main/java/org/dice_research/ldcbench/nodes/http/simple/dump/DumpFileResource.java

```diff
@@ -33,19 +36,25 @@
     public static final List<Archiver> ARCHIVERS = Arrays.asList(new TarArchiver(),
             new TarArchiver(ReflectionBasedStreamFactory.create("java.util.zip.GZIPOutputStream", "application/gzip", ".gz")),
             new ZipArchiver());
+    public static final ArrayList<File> multipleFile = new ArrayList<File>();
 
     public static DumpFileResource create(int domainId, String[] resourceUriTemplates, String[] accessUriTemplates,
-            Graph[] graphs, Predicate<Request> predicate, Lang lang, CompressionStreamFactory compression, Archiver archiver) {
+            Graph[] graphs, Predicate<Request> predicate, Lang lang, CompressionStreamFactory compression, boolean multipleDump, Archiver archiver) {
         DumpFileBuilder builder = new DumpFileBuilder(domainId, resourceUriTemplates, accessUriTemplates, graphs,
-                lang, compression);
+                lang, compression,multipleDump);
         try {
             File dumpFile = builder.build();
             String contentType = builder.buildContentType();
+            multipleFile.add(dumpFile);
             if (archiver != null)  {
-                //TODO support more than one file
-                //Add dump Files to a List and put them into Archive
+            	//Add dump Files to a List and put them into Archive
+            	Iterator<File> it = multipleFile.iterator();
                 File archive = File.createTempFile("ldcbench", ".archive");
-                archiver.buildArchive(archive,dumpFile);
+                ArchiveOutputStream aos =  archiver.createStream(archive);
+                while(it.hasNext()) {
+                	archiver.addFileToArchive(aos, it.next());
+                }
+            	contentType = archiver.getMediaType();
```

Comment:
I really don't understand how that should work. How does the class know how many of the graphs it should cover? Or will there be a single DumpFileResource for each graph? But if this is the case, why you had to change the class at all? :confused: